lib/core/src/Cardano/Wallet/Primitive/AddressDerivation.hs

```diff
@@ -722,3 +728,28 @@ hex = convertToBase Base16
 -- | Decode a 'ByteString' from base16
 fromHex :: ByteArray bout => ByteString -> Either String bout
 fromHex = convertFromBase Base16
+
+-- | Extracts the raw xprv bytes as @private key <> chain code@, i.e. without
+-- the public key.
+unXPrvStripPub :: XPrv -> ByteString
+unXPrvStripPub = stripPub . unXPrv
+  where
+    -- Converts  prv <> pub <> cc
+    -- To        prv <>        cc
+    stripPub :: ByteString -> ByteString
+    stripPub xprv' = prv <> chainCode
+      where
+        (prv, rest) = BS.splitAt 64 xprv'
+        (_pub, chainCode) = BS.splitAt 32 rest
+
+-- Create a @XPrv@ from a prv <> cc bytestring.
+--
+-- TODO: I'm not quite sure how this works, despite rountrip test passing.
+-- Sanity check.
+xPrvFromStrippedPubXPrv :: ByteString -> Either String XPrv
```

Comment:
This confused me:
```haskell
> bytes = "(\134\242|I\141L\EM\NUL\128\173\252q\191\172\167>f \218\222\167.\136\DC4\216\191\253r8cD8&I\STX;\185&\177\172E\241\185\241\157\226\r\163+\EM\GS\232-\188\250[E^N\129J\158\STX\134\188!\241\DLEzt\222\199\247U\143\ETB\128,\226Q\"\230\234\"\191\177\250\230\167\n\214X\244z\\"
> f = unXPrvStripPub . (either error Prelude.id . xPrvFromStrippedPubXPrv)
> f (BS.take 2 bytes) == bytes
True
-- However:
> BS.take 2 bytes
"(\134"
> f "(\134" == bytes
False
```
At first it seemed like I was magically able to re-create the XPrv from 2 bytes. But then it seemed like there is some behind the scenes magic related to `BS.take`.

Reply:
This is interesting also 😅 
```haskell
>>> unXPrv <$> xPrvFromStrippedPubXPrv (BS.take 0 bytes)
fish: 'stack ghci cardano-wallet-core' terminated by signal SIGSEGV (Address boundary error)
```

Reply:
https://github.com/input-output-hk/cardano-crypto/issues/67

Reply:
@Anviking lazyness maybe? Nothing tells you that you have to reduce your bytestring to WHNF for comparing them, perhaps the underlying implementation looks at length first, which is known to be different, so the Bytestring has to be different :man_shrugging: